tests/pytorch/test_numerics.py

```diff
@@ -354,6 +357,40 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:
         return (input > 0) * input * input
 
 
+class TorcGroupedLinearWithPadding(nn.Module):
+
+    def __init__(
+        self, num_gemms, in_features, out_features, bias, params_dtype, parallel_mode, fp8
+    ) -> None:
+        super().__init__()
+
+        self.padding = Fp8Padding(num_gemms)
+        self.linear_fn = GroupedLinear(
+            num_gemms,
+            in_features,
+            out_features,
+            bias=bias,
+            params_dtype=params_dtype,
+            parallel_mode=parallel_mode,
+            device="cuda",
+        )
+        self.unpadding = Fp8Unpadding(num_gemms)
```

Comment:
A question here, this GroupedLinear is used for MoE only. If padding for FP8 increases 2% E2E performance, why don't we have this padding embedded in GroupedLinear by default?

Reply:
In MLP Layer we have two successive GropedLinear if padding as a part of GroupedLinear we need paddding twice.

Reply:
Do we always call `moe_permutate` -> 2 x GroupedLinear -> `moe_unpermute`?

Reply:
Yes. In struct of transformer model MLP layer always has 2*Linear.

Reply:
Then, why don't we do FP8 padding in the permutation directly?

Reply:
I already have evaluated this solution but in the permutation of MCore it have multi code path.
This is a good idea. Maybe we can try it later.